webpack/assets/javascripts/foreman_class_edit.js

```diff
@@ -0,0 +1,201 @@
+import { sprintf } from './react_app/common/I18n';
+
+export function filterPuppetClasses(item) {
+  const term = $(item)
+    .val()
+    .trim();
+  const classElems = $('.available_classes').find(
+    '.puppetclass_group, .puppetclass'
+  );
+  if (term.length > 0) {
+    classElems
+      .hide()
+      .has(`[data-class-name*="${term}"]`)
+      .show();
+  } else {
+    classElems.show();
+  }
+}
+
+export function addPuppetClass(item) {
+  const id = $(item).attr('data-class-id');
+  const type = $(item).attr('data-type');
+  $(item).tooltip('hide');
+  const content = $(item)
+    .closest('li')
+    .clone();
+  content.attr('id', `selected_puppetclass_${id}`);
+  content.append(
+    `<input id='${type}_puppetclass_ids_' name='${type}[puppetclass_ids][]' type='hidden' value=${id}>`
+  );
+
+  const link = content.children('a');
+  const links = content.find('a');
+
+  links.attr('onclick', 'removePuppetClass(this)');
```

Comment:
Getting error on puppet class removal, change to `tfm.classEditor.removePuppetClass(this)` fixes the problem.

![remove-class](https://user-images.githubusercontent.com/2664090/56113120-34909a80-5f5d-11e9-930f-660fb7bfac2c.png)